backend/storage/backends.py

```
@@ -0,0 +1,199 @@
+                         local_path, exc_info=True)
+        return f"{_R2_PREFIX}{key}"
+
+    def load_json(self, ref: str) -> dict[str, Any]:
+        obj = self.client.get_object(Bucket=self.bucket, Key=self._key(ref))
+        return json.loads(obj["Body"].read().decode("utf-8"))
+
+    def read_bytes(self, ref: str) -> tuple[bytes, str] | None:
+        try:
+            obj = self.client.get_object(Bucket=self.bucket, Key=self._key(ref))
+            return obj["Body"].read(), obj.get("ContentType", "application/octet-stream")
+        except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)